packages/dashboard/src/components/composed/RPCs/RPC/Details/Expanded/DecodedParams.tsx

```diff
@@ -0,0 +1,49 @@
+import { Group, Text, createStyles } from "@mantine/core";
+import { Prism } from "@mantine/prism";
+import { AlertTriangle } from "react-feather";
+import type { DecodableRpcMethod } from "src/utils/constants";
+
+const useStyles = createStyles((theme, _params, _getRef) => ({
+  alert: {
+    color: theme.colors.orange[6]
+  }
+}));
+
+type DecodedParamsProps = {
+  decodingInspected: string;
+  decodingSucceeded: boolean;
+  method: DecodableRpcMethod;
+};
+
+function DecodedParams({
+  decodingInspected,
+  decodingSucceeded,
+  method
+}: DecodedParamsProps): JSX.Element {
+  const { classes } = useStyles();
+
+  const language =
+    method === "eth_sendTransaction"
+      ? "javascript"
+      : method === "eth_signTypedData_v3" || method === "eth_signTypedData_v4"
+      ? "json"
+      : "markdown";
+
+  return (
+    <>
+      <Group spacing={6}>
+        <Text size="sm" color="teal" weight={700}>
+          Decoded parameters
+        </Text>
+        {!decodingSucceeded && (
+          <AlertTriangle size={16} className={classes.alert} />
+        )}
+      </Group>
+      <Prism language={language} copyLabel="Copy to clipboard">
```

Comment:
Couple things:

- Is the coloring we get here worth including all of Prism?
- Note we use Highlightjs in other places because @haltman-at is the maintainer of highlightjs-solidity, so we've kind of coalesced around that highlighting system instead of Prism.
- What's the value of copying this decoded representation to the clipboard? Won't this go away when we have codec-components?

Reply:
Ah prism was added back in #5241, I wasn't aware that we were using highlightjs somewhere else (and our involvement with it). I picked it because it's provided by mantine (`@mantine/prism`) so it's in-line with other components in terms of props it accepts and styling.

Yes this will leave when codec-components come. With the button you can click once and send decodings to your friends :) jk this `copyLabel` prop just changes what shows on hover, the copy button is there by default.

I've noted this down as follow-up work! Makes sense to align with the rest of the monorepo and switch prism out.